File: lab8 interface/lab8 interface/Euro.cs
using System;

namespace OOP_Lab9_Variant19_Interface
{
    public class Euro : ICurrency, IPrintable
    {
        private double amount;
        private const double rate = 42.0;

        public Euro(double amount)
        {
            this.amount = amount;
        }

        public double ToUAH()
        {
            return amount * rate;
        }

        public void Print()
        {
            Console.WriteLine($"Euro: {amount} => {ToUAH()} UAH");
        }
    }
}

File: lab9/Euro.java
class Euro extends Currency {
    public Euro(double amount) {
        super(amount);
    }

    @Override
    public double toUAH() {
        return amount * 42;
    }

    @Override
    public void print() {
        System.out.println(amount + " Euro = " + toUAH() + " UAH");
    }
}
